### PR TITLE
Improve test output for multiple tests

### LINKED ISSUE #59177: Shouldn't a non-miri flag be added for syntax-extension
This is an excerpt from [syntax-extension-cfg](https://github.com/rust-lang/rust/blob/master/src/test/run-pass/macros/syntax-extension-cfg.rs)
```
// compile-flags: --cfg foo --cfg qux="foo"
// check <- Add #[cfg(not(miri))] // Miri does not support panics
pub fn main() {
    
    
    if ! cfg!(foo) { panic!() }
    if   cfg!(not(foo)) { panic!() }
    if ! cfg!(qux="foo") { panic!() }
    if   cfg!(not(qux="foo")) { panic!() }
    if ! cfg!(all(foo, qux="foo")) { panic!() }
// -- Omitted for brevity ---
```
Shouldn't the flag, as mentioned above be added (by the check comment)?

### PR DESCRIPTION
**Pull request by ohsayan**
This replaces several `assert!`s with `assert_eq!`s and `assert_ne!`s for better debugging and readability

### CODE REVIEW

File: src/test/run-pass/macros/syntax-extension-cfg.rs [hunk 1]
```diff
@@ -1,6 +1,7 @@
 // run-pass
 // compile-flags: --cfg foo --cfg qux="foo"
 
+#[cfg(not(miri))] // Miri does not support panics
```
Comment (oli-obk) [inline src/test/run-pass/macros/syntax-extension-cfg.rs]: it does support panics, it just doesn't support `should_panic` tests or `catch_panic`

Comment (sntdevco) [reply src/test/run-pass/macros/syntax-extension-cfg.rs]: Oh okay, will push now

Comment (sntdevco) [reply src/test/run-pass/macros/syntax-extension-cfg.rs]: @oli-obk Review most recent push, reverted the miri commit

File: src/test/run-pass/raw-fat-ptr.rs [hunk 2]
```diff
@@ -8,22 +8,22 @@ fn assert_inorder<T: PartialEq + PartialOrd>(a: &[T]) {
             if i < j {
                 assert!(a[i] < a[j]);
                 assert!(a[i] <= a[j]);
-                assert!(!(a[i] == a[j]));
-                assert!(a[i] != a[j]);
+                assert_eq!(!(a[i], a[j]));
```
Comment (mati865) [inline src/test/run-pass/raw-fat-ptr.rs]: This should be `assert_ne` and without `!` inside `()`.

**Pull request closed without merging**